### PR TITLE
Add mergeable config to check PR labels/milestones

### PR DESCRIPTION
**Pull request by yol**
We should set labels and milestone on GitHub PRs before merging for release management and regression/bug chasing.

It is easy to forget though so I propose to configure the mergeable but (already added to the repo) to check this for us.

This PR is the config for the bot.

### CODE REVIEW

**Review by DaveTBlake — APPROVED** [review]
Thanks @yol , I wonder what @Rechi is going to do with his spare time :)

**Review by Rechi — COMMENTED** [review]
`Abandoned, obsolete or superseeded` milestone shouldn't be allowed.
`Don't merge`, `No Jenkins`, `On hold`, `RFC` and `WIP` labels also should not be accepted.
Maybe you can add the configuration for disallowing `[RFC]` or `[WIP]` in the PR title.

**Comment by yol** [comment]
I agree with most things, buT:

> RFC and WIP labels also should not be accepted.
> Maybe you can add the configuration for disallowing [RFC] or [WIP] in the PR title.

I think that WIP PRs should be used for this, which do not allow merging anyway.

File: .github/mergeable.yml
Comment (yol) [reply .github/mergeable.yml]: Yes of course, thanks

**Comment by yol** [comment]
@Rechi updated - "No Jenkins" label not included since I think there are cases where you don't need Jenkins to build for merging, such as build documentation. Everything else should be in.

It at least seems to work now.

**Comment by Rechi** [comment]
Jenkins knows without telling if a PR includes only documentation changes or other changes that do not need CI, and in that case skips building that PR.
So in my opinion only Jenkins should decide if a PR needs building.

**Comment by DaveTBlake** [comment]
>So in my opinion only Jenkins should decide if a PR needs building.

Sometimes the dev knows best. Maybe don't want to waste resources tying up jenkins building at the time the PR is raised.

I think having something to remind us to set lables and milestones is a great idea,  but increasing the logic beyond that is unnecessary and possibly too limiting. Surely  if someone has the authority to merge a PR then they also have the intelligence to set those values suitably. Let's not over complicate this please.

**Comment by Rechi** [comment]
I wouldn't say it makes it more complicate. Actually you now have to look is the `No Jenkins` label set, why was it set and is it okay that the PR changes are merged without running a Jenkins PR build, before merging a PR.
Furthermore the `Mergeable` status causes GitHub to show `All checks have passed` if `No Jenkins` label is set (and everything else is fine). That might give the impression that the PR is ready to merge, while it's actually not.

As already said that's my opinion and if you think it's more useful to allow manually disabling the Jenkins build, I do not block it.

**Comment by yol** [comment]
> Jenkins knows without telling if a PR includes only documentation changes or other changes that do not need CI, and in that case skips building that PR.

Cool, I actually didn't know that! In that vein checking the "No Jenkins" label makes a lot more sense.

> Maybe don't want to waste resources tying up jenkins building at the time the PR is raised.

That's a very valid use of the label. Still, in that case, I think you should remove the label before merging the PR to make sure it went through CI once.

> Surely if someone has the authority to merge a PR then they also have the intelligence to set those values suitably.

I'm not sure if that's clear, but you will still be able to merge PRs that deviate from the "rules"/recommendations we set here. We do not currently have check passing set as mandatory for the repository, and it will probably stay that way. So it is just one more hint to make you double-check whether the PR is ready. If you disagree with that assessment, you can still merge - and update the mergeable config to account for the edge case we missed here ;-)

In the end, I don't want to make this a bikeshedding PR. I think anything is better than what we currently have. I also won't block over this small issue. So @DaveTBlake just give a final :+1:  or :-1: on the `No Jenkins` label and then I will go forward and merge this.

**Comment by DaveTBlake** [comment]
I already approved @yol hit that button!

**Comment by Rechi** [comment]
Don't forget to squash the commits.

**Comment by yol** [comment]
Alright!

**Pull request merged**
